lib/wallet/send_on_chain.dart

```diff
@@ -74,31 +74,28 @@ class _SendOnChainState extends State<SendOnChain> {
                       children: [
                         Row(
                           mainAxisAlignment: MainAxisAlignment.end,
-                          children: [
-                            ElevatedButton(
-                                onPressed: () async {
-                                  FLog.info(
-                                      text: "Sending " + amount.toString() + " sats to " + address);
-                                  String? error;
-                                  try {
-                                    await api.sendToAddress(address: address, amount: amount);
-                                  } on FfiException catch (e) {
-                                    error = e.message;
-                                  }
-
-                                  ScaffoldMessenger.of(context).showSnackBar(SnackBar(
-                                    content: Text(error == null
-                                        ? "Sent ${amount.toString()} satoshi to $address on-chain"
-                                        : "Failed to send bitcoin on-chain: $error"),
-                                  ));
-                                  context.go('/');
-                                },
-                                child: const Text('Send'))
-                          ],
+                          children: [AsyncButton(onPressedFunction: sendOnChain, label: "Send")],
                         ),
                       ],
                     ),
                   ),
                 ]))));
   }
+
+  void sendOnChain() {
+    FLog.info(text: "Sending $amount to $address");
+    api.sendToAddress(address: address, amount: amount).then((value) async {
```

Comment:
❓ Shouldn't we add an await here somewhere?

Reply:
With this syntax it is not needed - but I'm happy to stick to `try ... await ... catch`. I thought that you introduced the `.then` syntax and gave it a try 😅